### PR TITLE
fix(timeout): Added feature flag for rollback timeout ui input.

### PR DESCRIPTION
**Pull request by DanielaS12**
Feature was tested in previous pr. Adding feature flag.

### CODE REVIEW

File: orca-clouddriver/src/main/java/com/netflix/spinnaker/orca/clouddriver/tasks/cluster/DetermineRollbackCandidatesTask.java [hunk 1]
```diff
@@ -79,6 +80,9 @@ public class DetermineRollbackCandidatesTask implements CloudProviderAware, Retr
   private static final TypeReference<List<ServerGroup>> listOfServerGroupsTypeReference =
       new TypeReference<>() {};
 
+  @Value("${rollback.timeout.enabled:false}")
```
Comment (kkotula) [inline orca-clouddriver/src/main/java/com/netflix/spinnaker/orca/clouddriver/tasks/cluster/DetermineRollbackCandidatesTask.java:83]: The name of this property could be confusing to users and developers. It looks like the timeout functionality would be disabled completely if the flag is set to false or not set at all. Would it make sense to rename it to something like `rollback.dynamic-timeout.enabled` ?

File: orca-clouddriver/src/main/java/com/netflix/spinnaker/orca/clouddriver/tasks/cluster/DetermineRollbackCandidatesTask.java [hunk 2]
```diff
@@ -107,8 +111,10 @@ public long getTimeout() {
   }
 
   public long getDynamicTimeout(StageExecution stage) {
-    if (stage.getContext().containsKey("rollbackTimeout")) {
-      return TimeUnit.MINUTES.toMillis((int) stage.getContext().get("rollbackTimeout"));
+    if (stageRollbackTimeout) {
```
Comment (kkotula) [inline orca-clouddriver/src/main/java/com/netflix/spinnaker/orca/clouddriver/tasks/cluster/DetermineRollbackCandidatesTask.java:114]: To improve expressiveness and simplicity, we could do something like:
``` java
var shouldUseStageRollbackTimeout = stageRollbackTimeout
 && stage.getContext().containsKey("rollbackTimeout");

if (shouldUseStageRollbackTimeout){
    return TimeUnit.MINUTES....
}
return getTimeout();
```

It's a matter of taste though, so feel free to ignore this comment.

File: orca-clouddriver/src/main/java/com/netflix/spinnaker/orca/clouddriver/tasks/cluster/DetermineRollbackCandidatesTask.java [hunk 3]
```diff
@@ -79,6 +80,9 @@ public class DetermineRollbackCandidatesTask implements CloudProviderAware, Retr
   private static final TypeReference<List<ServerGroup>> listOfServerGroupsTypeReference =
       new TypeReference<>() {};
 
+  @Value("${rollback.timeout.enabled:false}")
+  private static boolean stageRollbackTimeout;
```
Comment (ovidiupopa07) [inline orca-clouddriver/src/main/java/com/netflix/spinnaker/orca/clouddriver/tasks/cluster/DetermineRollbackCandidatesTask.java:84]: doesn't need to be static. you can add a new argument to the existing constructor
```
private final boolean dynamicRollbackTimeoutEnabled;

  @Autowired
  public DetermineRollbackCandidatesTask(
      ObjectMapper objectMapper,
      RetrySupport retrySupport,
      CloudDriverService cloudDriverService,
      FeaturesService featuresService,
      @Value("${rollback.timeout.enabled:false}") boolean dynamicRollbackTimeoutEnabled) {
    this.retrySupport = retrySupport;
    this.cloudDriverService = cloudDriverService;
    this.previousImageRollbackSupport =
        new PreviousImageRollbackSupport(
            objectMapper, cloudDriverService, featuresService, retrySupport);
    this.dynamicRollbackTimeoutEnabled = dynamicRollbackTimeoutEnabled;
  }
  ```

**Review by ovidiupopa07 — CHANGES_REQUESTED** [review]
Let's also add a unit test to validate the feature flag.
One test to validate the functionality if it is enabled and one to check the functionality when it is disabled

**Comment by ovidiupopa07** [comment]
And I think we should also document how to enable this feature on the [docs page](https://github.com/spinnaker/spinnaker.io/blob/master/content/en/docs)

**Review by ovidiupopa07 — APPROVED** [review]
LGTM

**Comment by DanielaS12** [comment]
@Mergifyio backport release-2.28.x

**Comment by DanielaS12** [comment]
@ovidiupopa07 or @dogonthehorizon please backport this pr to 1.28.x

**Comment by ovidiupopa07** [comment]
> @ovidiupopa07 or @dogonthehorizon please backport this pr to 1.28.x

waiting for a confirmation from @dogonthehorizon . If everything ok I will backport to 1.27.x, 1.28.x and 1.29.x

**Comment by ovidiupopa07** [comment]
@mergify backport release-1.27.x release-1.28.x release-1.29.x

**Pull request merged**